pkg/crosscluster/replicationutils/utils.go

```diff
@@ -260,6 +262,65 @@ func ResolveHeartbeatTime(
 	return newProtectAbove
 }
 
+func GetAlterConnectionChecker(
+	id jobspb.JobID,
+	initiaDetails jobspb.Details,
+	uriGetter URIGetter,
+	execCfg *sql.ExecutorConfig,
+	stopper chan struct{},
+) func(ctx context.Context) error {
+	return func(ctx context.Context) error {
```

Comment:
There is a buglet I noticed when I was fixing the test flake that involved the connection poller. I think we should consider fixing now that its been extracted into a helper. 

We currently have a race condition:
1. We create the client for URI a.
2. The connection is changed to URI b.
3. We start the watcher wait for it to change from URI b.

Fixing this isn't too difficult. There are two options:
1. GetAlterConnectionChecker should accept the initial URI as an argument. 
2. We should call resolveURI in the calling goroutine before returning the function that is run as a polling goroutine. We should call `GetAlterConnectionChecker` before we create the stream client.